packages/create-subscription/src/__tests__/createSubscription-test.internal.js

```diff
@@ -0,0 +1,420 @@
+    ReplaySubject = require('rxjs/ReplaySubject').ReplaySubject;
+  });
+
+  function createBehaviorSubject(initialValue) {
+    const behaviorSubject = new BehaviorSubject();
+    if (initialValue) {
+      behaviorSubject.next(initialValue);
+    }
+    return behaviorSubject;
+  }
+
+  function createReplaySubject(initialValue) {
+    const replaySubject = new ReplaySubject();
+    if (initialValue) {
+      replaySubject.next(initialValue);
+    }
+    return replaySubject;
+  }
+
+  it('supports basic subscription pattern', () => {
+    const Subscription = createSubscription({
+      getValue: source => source.getValue(),
+      subscribe: (source, callback) => {
+        const subscription = source.subscribe(callback);
+        return () => subscription.unsubscribe;
+      },
+    });
+
+    const observable = createBehaviorSubject();
+    ReactNoop.render(
+      <Subscription source={observable}>
+        {(value = 'default') => {
+          ReactNoop.yield(value);
+          return null;
+        }}
+      </Subscription>,
+    );
+
+    // Updates while subscribed should re-render the child component
+    expect(ReactNoop.flush()).toEqual(['default']);
+    observable.next(123);
+    expect(ReactNoop.flush()).toEqual([123]);
+    observable.next('abc');
+    expect(ReactNoop.flush()).toEqual(['abc']);
+
+    // Unmounting the subscriber should remove listeners
+    ReactNoop.render(<div />);
+    observable.next(456);
+    expect(ReactNoop.flush()).toEqual([]);
+  });
+
+  it('should support observable types like RxJS ReplaySubject', () => {
+    const Subscription = createSubscription({
+      getValue: source => {
+        let currentValue;
+        source
+          .subscribe(value => {
+            currentValue = value;
+          })
+          .unsubscribe();
+        return currentValue;
+      },
+      subscribe: (source, callback) => {
+        const subscription = source.subscribe(callback);
+        return () => subscription.unsubscribe;
+      },
+    });
+
+    function render(value = 'default') {
+      ReactNoop.yield(value);
+      return null;
+    }
+
+    const observable = createReplaySubject('initial');
+
+    ReactNoop.render(<Subscription source={observable}>{render}</Subscription>);
+    expect(ReactNoop.flush()).toEqual(['initial']);
+    observable.next('updated');
+    expect(ReactNoop.flush()).toEqual(['updated']);
+
+    // Unsetting the subscriber prop should reset subscribed values
+    ReactNoop.render(<Subscription>{render}</Subscription>);
+    expect(ReactNoop.flush()).toEqual(['default']);
+  });
+
+  describe('Promises', () => {
+    it('should support Promises', async () => {
+      const Subscription = createSubscription({
+        getValue: source => undefined,
+        subscribe: (source, callback) => {
+          source.then(value => callback(value), value => callback(value));
+          // (Can't unsubscribe from a Promise)
+          return false;
```

Comment:
Maybe you should always return an unsubscribe function? If it's not supported, you can return a noop. By supporting a boolean, it implies that either option is equally fine, whereas you should always unsubscribe if possible.

Reply:
Oh I guess I missed the earlier comments about this. Put me down as +1 for always requiring an unsubscribe function. The fact that it's weirder in the promise case is a feature, IMO, not a bug.

Reply:
Yeah, I was on the fence about this. Returning `false` was a suggestion above but I'm okay with the noop approach. 👍

Reply:
I'm also fine with the no-op approach (I suggested `false` above).